staging/src/k8s.io/apiserver/pkg/storage/names/generate.go

```diff
@@ -41,9 +42,8 @@ var SimpleNameGenerator NameGenerator = simpleNameGenerator{}
 
 const (
 	// TODO: make this flexible for non-core resources with alternate naming rules.
-	maxNameLength          = 63
 	randomLength           = 5
-	maxGeneratedNameLength = maxNameLength - randomLength
+	maxGeneratedNameLength = validation.DNS1123SubdomainMaxLength - randomLength
```

Comment:
changing this globally for all consumers of this generator allows them to generate names that are invalid for some resources.

if we want a version of the name generator that allows longer names, make a `NewSimpleNameGenerator(maxNameLength int)` variant that individual rest resources can make use of with their specific max length

Reply:
I agree with you, although it won't cause any problems at present. I'll just change it where the pod name is generated.